profiles/common/modules/custom/nexteuropa_varnish/nexteuropa_varnish.module

```diff
@@ -223,6 +223,110 @@ function nexteuropa_varnish_node_presave($node) {
   }
 }
 
+/**
+ * Implements hook_node_delete().
+ */
+function nexteuropa_varnish_node_delete($node) {
+  // No rule defined for the node: Ignore the rest of the process.
+  if (!nexteuropa_varnish_node_has_rules($node)) {
+    return;
+  }
+
+  $paths = array();
+
+  $entity_uri = entity_uri('node', $node);
+
+  if ($entity_uri) {
+    $url_locale = language_negotiation_get('language', 'locale-url');
+    $url_suffix = language_negotiation_get('language', 'nexteuropa_multilingual_url_suffix');
+    // Add logic to send a language prefix/suffix regex for non aliased paths.
+    if ($url_locale) {
+      $lgregex = '[a-z]{2}\/';
+    }
+    elseif ($url_suffix) {
+      $lgregex = '_[a-z]{2}';
+    }
+    else {
+      $lgregex = '';
+    }
+    $original_path = $entity_uri['path'];
+    $alias = drupal_get_path_alias($original_path);
+    // Get the node\/nid.
+    $paths['unaliased'] = preg_quote($original_path, '/');
+    // Add regex prefix or suffix if needed.
+    // Example node\/nid_[a-z]{2}.
+    $paths['full_unaliased'] = $url_suffix ? $paths['unaliased'] . $lgregex : $lgregex . $paths['unaliased'];
+    // Add escaped alias.
+    // Example content\/my\-page.
+    $paths['escaped_alias'] = preg_quote($alias, '/');
+    // Add regex prefix or suffix if needed to alias.
+    // Example content\/my\-page_[a-z]{2}.
+    $paths['alias_prefix'] = $url_suffix ? $paths['escaped_alias'] . $lgregex : $lgregex . $paths['escaped_alias'];
+  }
+
+  // Add custom purge rules.
+  $purge_rules = nexteuropa_varnish_get_node_purge_rules($node);
+  foreach ($purge_rules as $purge_rule) {
+    if ($purge_rule->type() == PurgeRuleType::PATHS) {
+      $purge_paths = $purge_rule->paths();
+      $paths = array_merge($paths, $purge_paths);
+    }
+  }
+
+  if (!empty($paths)) {
+    // Removing duplicates and all entries of the array equal to FALSE.
+    $paths = array_unique($paths);
+    $paths = array_filter($paths);
+    _nexteuropa_varnish_purge_paths($paths);
+  }
+}
+
+/**
+ * Implements hook_path_delete().
+ */
+function nexteuropa_varnish_path_delete($path) {
+  if (empty($path)) {
+    return;
+  }
+
+  $node = menu_get_object('node', 1, $path['source']);
+
+  // Its a node path.
+  if (!empty($node)) {
```

Comment:
Maybe an early return here would be better, just like you did with `$path`.

```php
if (empty($node)) {
  return;
}
```

It lower the complexity of the code, it's consistent with the rest, and the main code doesn't gets wrapped by a condition.

Reply:
Fixed.